src/blockdata/transaction.rs

```diff
@@ -406,15 +406,20 @@ impl Transaction {
 
     #[cfg(feature="bitcoinconsensus")]
     /// Verify that this transaction is able to spend its inputs
-    /// The lambda spent should not return the same TxOut twice!
-    pub fn verify<S>(&self, mut spent: S) -> Result<(), script::Error>
-        where S: FnMut(&OutPoint) -> Option<TxOut> {
-        let tx = serialize(&*self);
+    pub fn verify<'a, S>(&self, get_utxo: S) -> Result<(), script::Error>
+        where S: Fn(OutPoint) -> Option<&'a TxOut> {
```

Comment:
Since you now call it only once, `FnOnce` would be more general and more appropriate.

Reply:
I never fully understood `Fn` vs `FnOnce`. You call the method for every input, right? So not once. Unless "once" here has some extra meaning I'm missing.

Reply:
FnOnce is a closure that *cannot* be called more than once, usually by moving a non Copy type from the outside environment to the closure.
example: https://play.rust-lang.org/?gist=9f86723ad4d2b0530a45d4047f311efc

Reply:
Oh, you're right, it's called in a loop. So you need to use `FnMut`.

Please do not use `Fn`. It is never necessary and always restrictive.